docs/develop.md

```diff
@@ -1,18 +1,426 @@
+initial efforts, in our experience this is the preferred way to authenticate
+with GitHub.
+
+There is also the option to use the GitHub command line interface. This is a
+[separate software](https://cli.github.com/) provided by GitHub. We have little
+experience with it.
+
+Once you've decided how to authenticate, copy the corresponding url. Then open a
+terminal and paste the url preceded with `git clone`. In case of using SSH:
+
+```
+# With fork
+git clone git@github.com:Peter9192/WAM2layers.git
+
+# Without fork
+git clone git@github.com:WAM2layers/WAM2layers.git
+```
+
+### Development installation - additional tooling
+
+You can install your local copy of the code by following the same steps as the
+[regular installation](./installation.md). However, by adding `[develop]`, some
+additional dependencies are installed that are convenient for developing.
+
+```
+pip install --editable .[develop]
+```
+
+The `--editable` flag means that any changes you make to the code will be
+"effective immediately". When you run the model again, the updated code is used.
+
+The `[develop]` option tells pip to install not just the code, but some
+additional packages that are listed under the "develop" header in the file
+`pyproject.toml`. These packages help with linting (checking your code against
+syntax/style guides), automatic formatting, building documentation, running
+tests, and publising the package on PyPI.
+
+### Create a new branch
+
+It is good practice to create a new "branch" for each feature you are adding.
+For example, this will create (and switch to) a new branch called
+"developer-documentation":
+
+```
+git checkout -b developer-documentation
+```
+
+You can use `git branch` to see all your branches, and
+`git checkout <branchname>` (without the `-b`) to switch to another (existing)
+branch. In case of the example, I would use this branch while working on this
+documentation page. But if I was interrupting my work to fix a bug on another
+part of the code, I would switch to another branch. This keeps all changes that
+belong together, well, together.
+
+### Make changes
+
+Now you are ready to make the changes you need and test whether they work. Keep
+in mind that your changes should not break the code for other people. If you're
+adding new code, please also add relevant documentation, and add an update to
+the changelog. If your introducing new dependencies, add them to
+`pyproject.toml`. For more information on the inner workings of WAM2layers, see
+below.
+
+### Follow the style guide
+
+Try to follow the Python style guide
+([pep8](https://peps.python.org/pep-0008/)). This is where the developer tools
+come in handy:
+
+```
+# flake8 can check your code against PEP8
+flake8 .
+
+# isort can automatically format import statements
+isort .
+
+# we use black for automatic formatting
+black .
+```
+
+### Commit and push your changes
+
+Next, you can commit them and push your branch to the remote:
+
+```
```

Comment:
maybe we can mention that this can also be done with a git extension in VScode or Github for Windows or...